deepspeed/utils/tensor_fragment.py

```diff
@@ -207,6 +216,26 @@ def safe_get_full_grad(param):
     return None
 
 
+def safe_set_full_grad(param, value):
+    """Update the partitioned gradient of a low-precision (e.g., fp16) parameter.
+
+        Args:
+            param (``torch.nn.Parameter``): A model parameter
+            value (``torch.Tensor``): New value
```

Comment:
Can we clarify `value` is a partitioned value for ZeRO3? I think we pass the entire grad for ZeRO1. How about ZeRO2?

Reply:
Great question! The general principle of these APIs is that output values returned by `get_` and input values of `set_` are the full unpartitioned value. This is because we don't want users to worry about partitioning when manipulating these tensors. I will carefully update the docs globally to provide this clarification.